Remotion/Validation/UnitTests/Rules/ObjectValidationRuleTest.cs

```diff
@@ -1,21 +1,132 @@
-﻿using NUnit.Framework;
+﻿using System;
+using System.Collections.Generic;
+using Moq;
+using NUnit.Framework;
+using Remotion.Validation.Results;
+using Remotion.Validation.Rules;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Rules
 {
   [TestFixture]
   public class ObjectValidationRuleTest
   {
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void Validate ()
+    public void Validate_ReturnsValidationFailures ()
     {
+      Func<int, bool> condition = i => true;
+      var context = new ValidationContext(1);
+
+      var failure1 = new ObjectValidationFailure(
+          15,
+          "15 is not a number",
+          "Localized validation message: 15 is not a number.");
+
+      var failure2 = new ObjectValidationFailure(
+          42,
+          "42 is not a number",
+          "Localized validation message: 42 is not a number.");
+
+      var validator1Mock = new Mock<IObjectValidator>();
+      validator1Mock
+          .Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new[] { failure1 })
+          .Verifiable();
+
+      var validator2Mock = new Mock<IObjectValidator>();
+      validator2Mock
+          .Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new[] { failure2 })
+          .Verifiable();
+
+      var rule = new ObjectValidationRule<int>(condition, new[] { validator1Mock.Object, validator2Mock.Object });
+
+      var result = rule.Validate(context);
+      Assert.That(result, Is.EquivalentTo(new[] { failure1, failure2 }));
+      validator2Mock.Verify();
+      validator1Mock.Verify();
```

Comment:
We don't need them verifiable, isntead stub semantics is sufficient because the result can only satisfy the assertion if both validators where called correctly. please also update the name to Stub